src/snyk/snykCode/error/snykCodeErrorHandler.ts

```diff
@@ -124,31 +145,38 @@ export class SnykCodeErrorHandler extends ErrorHandler implements ISnykCodeError
     callback: (error: errorType) => void,
   ): void {
     this.transientErrors = 0;
-    this._requestId = undefined;
-
     callback(error);
+
     this.capture(error, options, { [TagKeys.CodeRequestId]: this._requestId });
+    this.resetRequestId();
   }
 
-  private async connectionErrorHandler(
+  private async retryHandler(
     error: errorType,
     options: { [key: string]: unknown },
     callback: (error: Error) => void,
   ): Promise<void> {
     this.logger.error(`Connection error to Snyk Code. Try count: ${this.transientErrors + 1}.`);
+
     if (this.transientErrors > MAX_CONNECTION_RETRIES) {
       this._connectionRetryLimitExhausted = true;
       return this.generalErrorHandler(error, options, callback);
     }
 
     this.transientErrors += 1;
+
     setTimeout(() => {
       this.baseSnykModule.runCodeScan().catch(err => this.capture(err, options));
     }, CONNECTION_ERROR_RETRY_INTERVAL);
+
     return Promise.resolve();
   }
 
   capture(error: errorType, options: { [key: string]: unknown }, tags?: Tags): void {
+    if (error instanceof SnykCodeErrorResponse) {
+      error = new Error(JSON.stringify(error?.error));
```

Comment:
Should we also have a custom error, e.g. "SnykCodeError" for ease of searching when errors get reported? Wdyt?

Reply:
Already discussed it. 

Essentially we can see that the error originates from `snykCodeErrorHandler` whenever it does and it might be good enough to identify and search for. Definitely tweak it in the future if it is still a pain.